gateway/middleware.go

```diff
@@ -31,10 +34,53 @@ var (
 
 func init() {
 	flag.StringVar(&jwtSecret, "gateway.auth.jwt-secret", "", "Secret to sign JSON Web Tokens")
+	flag.BoolVar(&jwtSecretEncoded, "gateway.auth.jwt-secret-encoded", false, "Whether to base64-decode secret before use")
+	flag.StringVar(&jwtValidationAlgo, "gateway.auth.jwt-validation-algo", SigningAlgoHMAC, "JWT Validation algorithm: HMAC or RSA")
 }
 
+
 // AuthenticateTenant validates the Bearer Token and attaches the TenantID to the request
 var AuthenticateTenant = middleware.Func(func(next http.Handler) http.Handler {
+
+	// Initialize JWT secret
+	jwtSecretBytes := []byte(jwtSecret)
+
+	// RSA Key
+	var jwtRsaKey rsa.PublicKey
+
+	var err error
+
+	// Optionally decode JWT secret
+	if jwtSecretEncoded {
+		secretDecoded, err := base64.StdEncoding.DecodeString(jwtSecret)
```

Comment:
The idea of base64 encoded secret here is that it's usually not a string, otherwise it can simply be decoded and provided in plain text in `gateway.auth.jwt-secret` value. Base64 encoded secret is basically an array of bytes encoded in base64 ([here](https://github.com/lcobucci/jwt/issues/217#issuecomment-354596032) is some explanation).
In most libraries I've seen base64url decoding is used for such cases, for example see how [Keycloak does that](https://github.com/keycloak/keycloak/blob/66dfa32cd569a7416de21b4dc04db212e8fccce5/core/src/main/java/org/keycloak/jose/jws/crypto/HMACProvider.java#L94-L101).